### PR TITLE
added search bar

### PR DESCRIPTION
**Pull request by Henrynunez112**
**Description**
Added a search bar to the connect modal.
issue #141

### CODE REVIEW

File: packages/use-contractkit/src/modals/connect.tsx [hunk 1]
```diff
@@ -148,10 +149,14 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
           ([providerKey, provider]) =>
             typeof window !== 'undefined' &&
             provider.showInList() &&
-            Object.keys(allScreens).find((screen) => screen === providerKey)
+            Object.keys(allScreens)
+              .filter((screen) =>
+                screen.toLowerCase().includes(search.toLowerCase())
+              )
```
Comment (aaronmgdr) [inline packages/use-contractkit/src/modals/connect.tsx:155]: what about cases where there is not a search yet?

Comment (aaronmgdr) [reply packages/use-contractkit/src/modals/connect.tsx:155]: This is filtering (line 153) inside a filter (line 148) which is less than optimal. 

the is probably a way to do this without the n squared filtering

Comment (nicolasbrugneaux) [reply packages/use-contractkit/src/modals/connect.tsx:155]: I agree this is not optimal, but we're talking about tiny amount of data here, we don't have 100s of screens nor 100s of providers

File: packages/use-contractkit/src/modals/connect.tsx [hunk 2]
```diff
@@ -148,10 +149,11 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
           ([providerKey, provider]) =>
             typeof window !== 'undefined' &&
             provider.showInList() &&
-            Object.keys(allScreens).find((screen) => screen === providerKey)
+            providerKey in allScreens &&
+            providerKey.toLowerCase().includes(search.toLowerCase())
         )
         .sort(([, a], [, b]) => sort(a, b)),
-    [_providers, allScreens, sort]
+    [_providers, allScreens, sort, search]
```
Comment (crondinini) [inline packages/use-contractkit/src/modals/connect.tsx:156]: As it is now, the search being inside this `useMemo` leads to a bug where if you close the modal after a search, that search will be saved but the text in the input is not up to date. 
- Control search separately from `providers`
or 
- Control the value for the input using the search value (probably should be done regardless)
- Reset the `search` state on close
 
Kapture 2022-04-28 at 11 17 28

Comment (crondinini) [inline packages/use-contractkit/src/modals/connect.tsx:156]: As it is now, the search being inside this `useMemo` leads to a bug where if you close the modal after a search, that search will be saved but the text in the input is not up to date. 
- Control search separately from `providers`
or 
- Control the value for the input using the search value (probably should be done regardless)
- Reset the `search` state on close
 
Kapture 2022-04-28 at 11 17 28

File: packages/use-contractkit/src/modals/connect.tsx [hunk 3]
```diff
@@ -173,6 +175,10 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
         <h1 className="tw-pl-3 tw-pb-2 tw-text-lg tw-font-medium dark:tw-text-gray-300">
           {title}
         </h1>
+        <input
+          placeholder="Search"
```
Comment (crondinini) [inline packages/use-contractkit/src/modals/connect.tsx:179]: I'd be more explicit here and say "Search for a wallet"

Comment (crondinini) [inline packages/use-contractkit/src/modals/connect.tsx:179]: I'd be more explicit here and say "Search for a wallet"

**Review by crondinini — COMMENTED** [review]
I'd say there are still some things to be addressed before this can be merged.

**Comment by aaronmgdr** [comment]
closing since this was merged into #169

**Pull request closed without merging**